### DISCUSSION TITLE
Announcement: An Upcoming Visual Studio Code Extension for Avalonia!

### DISCUSSION THREAD

**Discussion by MikeCodesDotNET** [AvaloniaUI/Avalonia#12128] [General]
We have some exciting news for you! We're in the process of developing a brand-new Visual Studio Code extension specifically tailored for Avalonia. This development is being led by [Prashant Cholachagudda](https://github.com/prashantvc), a seasoned veteran and former Microsoft and Xamarin employee!

The extension will significantly enhance the Avalonia development experience in Visual Studio Code. It features our XAML code completion and integrated XAML previewer, making it easier than ever to build beautiful Avalonia apps. And the best part? It will be released as open-source, free for everyone to use and contribute to in the coming months.

In appreciation of our GitHub sponsors' invaluable support, we will offer them early access (very soon). This is our way of saying thank you for their dedication and contribution to the Avalonia project. If you're eager to try it before the public release, consider becoming a sponsor. It's a great way to support the project and get exclusive perks!

Once again, thank you to all contributors and our community for your continued support! If you want to sponsor the project, please check out our [GitHub Sponsors page](https://github.com/sponsors/AvaloniaUI). Your contributions truly make a difference.

Here's an early sneak peek of the extension in action! 
[![Visual Studio Extension - Sneak Peek](http://img.youtube.com/vi/UaMbjY0HX4E/0.jpg)](http://www.youtube.com/watch?v=UaMbjY0HX4E "Visual Studio Extension - Sneak Peek")

**Comment by timunie** [comment]
Incredible!

**Comment by rabbitism** [comment]
Can't wait to try it!

**Comment by florianwachs** [comment]
just wow!

**Comment by danielmayost** [comment]
Eagerly waiting!

**Comment by SeryiBaran** [comment]
Wow 😍

**Comment by muhammadkhon9711** [comment]
excellent!

**Comment by scriptBoris** [comment]
Awesome!

**Comment by prashantvc** [comment]
FYI https://github.com/AvaloniaUI/Avalonia/discussions/12337
